File: design-patterns/go/bridge/ui_element/input.go
// input.go

package main

import "fmt"

type Input struct {
	colorSchema ColorSchema
}

func NewInput(colorSchema ColorSchema) (input *Input) {
	input = &Input{}
	input.colorSchema = colorSchema
	return
}

func (input *Input) PrintElement() {
	input.colorSchema.SetColor()

	fmt.Println("Printing Input")
}

File: design-patterns/php/bridge/uielement/Input.php
<?php
// Input.php

namespace BigBoxCode\DesignPattern\Bridge\UiElement;

class Input extends UIElement {
    public function __construct(Color $color) {
        parent::__construct($color);
    }

    public function printElement(): void {
        $this->color->setColor();

        echo "Printing Input\n";
    }
}
